pkg/kubelet/volumemanager/reconciler/reconciler.go

```diff
@@ -448,15 +482,51 @@ func (rc *reconciler) reconstructVolume(volume podVolume) (*reconstructedVolume,
 		uniqueVolumeName = volumehelper.GetUniqueVolumeNameForNonAttachableVolume(volume.podName, plugin, volumeSpec)
 	}
 
-	if attachablePlugin != nil {
-		if isNotMount, mountCheckErr := rc.mounter.IsLikelyNotMountPoint(volume.mountPath); mountCheckErr != nil {
-			return nil, fmt.Errorf("Could not check whether the volume %q (spec.Name: %q) pod %q (UID: %q) is mounted with: %v",
-				uniqueVolumeName,
-				volumeSpec.Name(),
-				volume.podName,
-				pod.UID,
-				mountCheckErr)
-		} else if isNotMount {
+	// TODO: remove feature gate check after no longer needed
+	if utilfeature.DefaultFeatureGate.Enabled(features.BlockVolume) {
+		// Check existence of mount point for filesystem volume or symbolic link for block volume
+		var isNotMount, islinkExist bool
+		var mountCheckErr, checkErr error
+		if attachablePlugin != nil && volume.volumeMode == v1.PersistentVolumeFilesystem {
```

Comment:
Why do we need to check for filesystem mount point under the block volume feature?  Can this also be part of the handler?

Reply:
ok. I think this check should be separated at block case and filesystem case.

Reply:
Even if we enable block volume feature, both Filesystem and Block is coming to reconstruct. Therefore, we should check before calling mount point check or symlink existing check. Yes, I think we could add this handler.

Reply:
done.